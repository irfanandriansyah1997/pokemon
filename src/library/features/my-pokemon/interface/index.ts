export * from './my-pokemon-hooks.interface';
export * from './register-dialog.interface';
