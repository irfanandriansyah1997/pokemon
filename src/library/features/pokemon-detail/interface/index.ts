export * from './pokemon-item.interface';
