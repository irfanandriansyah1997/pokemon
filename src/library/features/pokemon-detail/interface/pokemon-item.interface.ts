/**
 * Pokemon Item Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonItem {
  id: number;
  image: string;
  nickname?: string;
}
