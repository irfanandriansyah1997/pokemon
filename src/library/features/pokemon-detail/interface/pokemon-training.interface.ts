/**
 * Pokemon Training Props Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.31
 */
export interface IPokemonTrainingProps {
  title: string;
  value: string;
}
