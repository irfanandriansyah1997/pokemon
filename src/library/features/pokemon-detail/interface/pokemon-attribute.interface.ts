import { IPokemonSpecies } from '@/library/interface/pokemon';

/**
 * Pokemon Attribute Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.31
 */
export interface IPokemonAttribute {
  key: string;
  value: string | string[];
}

/**
 * Pokemon Attribute Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.31
 */
export interface IPokemonAttributeParameter {
  height: number;
  pokeSpecies?: IPokemonSpecies;
  weight: number;
}
