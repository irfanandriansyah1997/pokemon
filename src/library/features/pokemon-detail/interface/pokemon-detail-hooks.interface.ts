import { IPokemon, IPokemonSpecies } from '@/library/interface/pokemon';

import { IPokemonDetailReducer } from './pokemon-detail-reducer.interface';

/**
 * Pokemon Detail Hooks
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonDetailHooks {
  action: IPokemonDetailHooksAction;
  state: IPokemonDetailReducer;
}

/**
 * Pokemon Detail Hooks Action Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonDetailHooksAction {
  setPokemon(param: IPokemon | undefined): void;
  setPokemonSpecies(param: IPokemonSpecies): void;
  toggleLoading(param: boolean): void;
}
