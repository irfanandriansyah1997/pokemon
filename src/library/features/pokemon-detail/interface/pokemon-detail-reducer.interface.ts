import { GenActionMap, NullAble } from '@/library/interface/general';
import { IBaseQueryError } from '@/library/interface/gql';
import { IPokemon, IPokemonSpecies } from '@/library/interface/pokemon';

/**
 * Pokemon Detail Reducer
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonDetailReducer {
  error?: IBaseQueryError;
  isLoadingRest: boolean;
  pokemon?: IPokemon;
  selection: number;
}

/**
 * Pokemon Detail Enum Reducers
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export enum IPokemonDetailTypesEnum {
  setLoading = `SET_LOADING`,
  setPokemonState = `SET_POKEMON`,
  setSelection = `SET_SELECTION`,
  setPokemonSpecies = `SET_POKEMON_SPECIES`
}

/**
 * Pokemon Detail Payload Action Reducers
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
type IPokemonDetailPayload = {
  [IPokemonDetailTypesEnum.setLoading]: boolean;
  [IPokemonDetailTypesEnum.setPokemonState]: NullAble<IPokemon>;
  [IPokemonDetailTypesEnum.setSelection]: number;
  [IPokemonDetailTypesEnum.setPokemonSpecies]: IPokemonSpecies;
};

export type IPokemonDetailActionType = GenActionMap<IPokemonDetailPayload>[keyof GenActionMap<IPokemonDetailPayload>];
