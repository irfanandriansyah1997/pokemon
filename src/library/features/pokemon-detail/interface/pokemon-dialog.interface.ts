import { IEventComponent } from '@/library/interface/general';
import { IPokemon, IPokemonSprites } from '@/library/interface/pokemon';

export type IEventOnClose = IEventComponent<'on-close'>;

export type IEventOnRelease = IEventComponent<'on-release'>;

export type IPokemonDialogEvent = (
  event: IEventOnClose | IEventOnRelease
) => void;

/**
 * Pokemon Dialog Props
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonDialogProps {
  on: IPokemonDialogEvent;
  pokemon?: IPokemon;
  showDialog: boolean;
}

/**
 * Pokemon Dialog Backdrop Props
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.31
 */
export interface IPokemonDialogBackdropProps {
  color?: string;
  show?: boolean;
}

/**
 * Pokemon Top Section Props
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.31
 */
export interface IPokemonTopSectionProps {
  genus: string;
  id: string;
  name: string;
  showImage?: boolean;
  showWrapper?: boolean;
  sprite?: IPokemonSprites;
}
