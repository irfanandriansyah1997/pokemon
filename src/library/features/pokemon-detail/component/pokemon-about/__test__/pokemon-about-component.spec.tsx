import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import PokemonAbout from '@/library/features/pokemon-detail/component/pokemon-about';
import FullPokemonFixture from '@/library/features/pokemon-detail/reducers/__test__/__fixture__/full-pokemon.fixture.json';
import { IPokemon } from '@/library/interface/pokemon';

describe(`Testing Pokemon About Component`, () => {
  it(`Testing Snapshot Test`, () => {
    const ui = shallow(
      <PokemonAbout {...((FullPokemonFixture as unknown) as IPokemon)} />
    );

    expect(toJson(ui)).toMatchSnapshot();
  });
});
