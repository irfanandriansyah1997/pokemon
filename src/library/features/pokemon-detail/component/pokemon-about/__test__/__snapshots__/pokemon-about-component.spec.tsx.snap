// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Pokemon About Component Testing Snapshot Test 1`] = `
<Fragment>
  <Styled(p)
    style={
      Object {
        "marginTop": 24,
      }
    }
    textAlign="justify"
  >
    BULBASAUR can be seen napping in
bright sunlight.
There is a seed on its back.By soaking up the sunâ€™s rays, the seed
grows progressively larger.
  </Styled(p)>
  <Styled(div)>
    <Styled(div)
      key="grass"
    >
      grass
    </Styled(div)>
    <Styled(div)
      key="poison"
    >
      poison
    </Styled(div)>
  </Styled(div)>
  <Styled(p)>
    Pokedex Data
  </Styled(p)>
  <Styled(div)
    rowGap={16}
    templates={
      Array [
        "90px",
        "5px",
        "auto",
      ]
    }
  >
    <Styled(p)
      fontSize="text"
      fontWeight={500}
    >
      Height
    </Styled(p)>
    <Styled(p)
      fontSize="text"
    >
      :
    </Styled(p)>
    <Styled(p)
      fontSize="text"
    >
      0.7 m
    </Styled(p)>
    <Styled(p)
      fontSize="text"
      fontWeight={500}
    >
      Weight
    </Styled(p)>
    <Styled(p)
      fontSize="text"
    >
      :
    </Styled(p)>
    <Styled(p)
      fontSize="text"
    >
      6.9 kg
    </Styled(p)>
    <Styled(p)
      fontSize="text"
      fontWeight={500}
    >
      Egg Cycle
    </Styled(p)>
    <Styled(p)
      fontSize="text"
    >
      :
    </Styled(p)>
    <Styled(p)
      fontSize="text"
    >
      20 Days
    </Styled(p)>
    <Styled(p)
      fontSize="text"
      fontWeight={500}
    >
      Egg Groups
    </Styled(p)>
    <Styled(p)
      fontSize="text"
    >
      :
    </Styled(p)>
    <Styled(div)>
      <Styled(p)
        fontSize="text"
        key="monster"
      >
        <img
          alt="monster"
          loading="lazy"
          src="eggs.svg"
        />
        monster
      </Styled(p)>
      <Styled(p)
        fontSize="text"
        key="plant"
      >
        <img
          alt="plant"
          loading="lazy"
          src="eggs.svg"
        />
        plant
      </Styled(p)>
    </Styled(div)>
  </Styled(div)>
  <Styled(p)>
    Ability
  </Styled(p)>
  <Styled(ul)>
    <li
      key="overgrow"
    >
      <Styled(p)
        fontSize="normal"
        fontWeight={500}
      >
        overgrow
         
      </Styled(p)>
    </li>
    <li
      key="chlorophyll"
    >
      <Styled(p)
        fontSize="normal"
        fontWeight={500}
      >
        chlorophyll
         
        (Hidden Ability)
      </Styled(p)>
    </li>
  </Styled(ul)>
  <Styled(p)>
    Training
  </Styled(p)>
  <Styled(div)>
    <PokemonTraining
      title="Base Experience"
      value="64"
    />
    <PokemonTraining
      title="Base Happiness"
      value="70"
    />
    <PokemonTraining
      title="Catch Rate"
      value="45%"
    />
    <PokemonTraining
      title="Growth Rate"
      value="medium-slow"
    />
  </Styled(div)>
</Fragment>
`;
